src/emulate.c

```diff
@@ -99,11 +89,9 @@ static void trap_handler(riscv_t *rv);
  * it is worth noting that a future standard could redefine how m/stval is
  * handled for different types of traps.
  *
- * For simplicity and clarity, abstracting stval and mtval into a single
- * identifier called tval, as both are handled by TRAP_HANDLER_IMPL.
  */
-#define TRAP_HANDLER_IMPL(type, code)                                         \
-    static void rv_trap_##type(riscv_t *rv, uint32_t tval)                    \
+#define TRAP_HANDLER_IMPL(type)                                               \
+    static void rv_trap_##type(riscv_t *rv)                                   \
```

Comment:
We do not need different handler for different trap type anymore. The information for trapping is stored in `m/scause`, `m/stval`.

Reply:
> We do not need different handler for different trap type anymore. The information for trapping is stored in `m/scause`, `m/stval`.

Could you elaborate more on this?

Reply:
> Could you elaborate more on this?

I meant we could simplify some handlers in `_trap_handler()` (e.g. illegal_insn, breakpoint, misalligned). They have the same routine of reading `m/stval` and `m/scause` to identify the type of traps.

Reply:
> > Could you elaborate more on this?
> 
> I meant we could simplify some handlers in `_trap_handler()` (e.g. illegal_insn, breakpoint, misalligned). They have the same routine of reading `m/stval` and `m/scause` to identify the type of traps.

It make sense.